test/test_torch.py

```diff
@@ -11378,6 +11378,53 @@ def test_sign(self, device):
         a_bool.sign_()
         self.assertEqual(a_bool, a_bool_target, msg='sign_ device={} dtype=bool'.format(device))
 
+    def test_signbit(self, device):
```

Comment:
This test is pretty good but I think it can be simplified by splitting it into cases. For example, ints (including bool), floats, and complex types. You can make separate tests for each ("test_signbit_complex", "test_signbit_int", ...) and use the `@dtypes` decorator to specify the dtypes each test wants. 

For the float test, you can use `torch.randn` to generate a random tensor and keep your tensor of interesting float values (which you can extend with +inf and -inf). Then you can use the `compare_with_numpy` function to verify that `torch.signbit` produces the same output as `np.signbit` on these tensors. (You'll need to add a decorator to skip the test if numpy isn't available, too.) 

For int values you can just use `torch.randint` to generate a random tensor, and for complex values you just want to verify a runtime error is thrown. 

After comparing with NumPy, you can test the out variant with the same input and verify the tensor populated in out is the same as that returned from signbit.

Does that make sense? I think that will make the tests simpler and more robust.

Reply:
I couldn't agree more. I have improved the test cases, split them into three parts, please make a check does it robust enough?